src/modules/extrakeys/hsort_impl.h

```diff
@@ -0,0 +1,100 @@
+/***********************************************************************
+ * Copyright (c) 2021 Russell O'Connor, Jonas Nick                     *
+ * Distributed under the MIT software license, see the accompanying    *
+ * file COPYING or https://www.opensource.org/licenses/mit-license.php.*
+ ***********************************************************************/
+
+#ifndef SECP256K1_HSORT_IMPL_H_
+#define SECP256K1_HSORT_IMPL_H_
+
+#include "hsort.h"
+
+/* An array is a heap when, for all non-zero indexes i,
+ * the element at index i compares as less than or equal to the element at index parent(i).
+ */
+
+/* Precondition: 2*i <= SIZE_MAX - 1 */
+static SECP256K1_INLINE size_t child1(size_t i) { return 2*i + 1; }
+
+/* Precondition: 2*i <= SIZE_MAX - 2 */
+static SECP256K1_INLINE size_t child2(size_t i) { return child1(i)+1; }
+
+/* Precondition: 0 < i */
+static SECP256K1_INLINE size_t parent(size_t i) { return (i-1)/2; }
```

Comment:
I don't know if this is a problem or not, but I did notice that `parent` seems to be only used in comments.

Reply:
Indeed, I think then we should remove it or better replace it by a comment.

Reply:
done